### DISCUSSION TITLE
Install ADempiere 3.9.4 on Ubuntu 22.04 as User "adempiere" with sudo access

### DISCUSSION THREAD

**Discussion by marcalwestf** [adempiere/adempiere#4061] [Show and tell]
(The following notes were provided by Paul Bowden (@pawlbouden) and Steven Sackett (@adaxa-steven)  from Adaxa. I just formatted the text and added some explanations)

### Install Dos2Unix
Dos2Unix is a package that contains dos2unix and unix2dos commands for converting line endings in a text file from DOS/Windows to Unix/Linux and vice versa.
- `cd /home/adempiere`
- `sudo apt install dos2unix` 

### Install JDK 11
Reminder: it can be any JDK 11.
Also possible is JDK 17.
- `cd /home/adempiere`
- `sudo apt install openjdk-11-jdk` 

### Install Postgres
- `sudo apt install postgresql`

### Deploy ADempiere 3.9.4
- `wget https://github.com/adempiere/adempiere/releases/download/3.9.4/Adempiere_394LTS.tar.gz`
- `tar xvf Adempiere_394LTS.tar.gz`

### Deploy Tomcat 9
- `wget https://dlcdn.apache.org/tomcat/tomcat-9/v9.0.71/bin/apache-tomcat-9.0.71.tar.gz`
- `tar xvf apache-tomcat-9.0.71.tar.gz`

#### Get Adempiere Server Properties File
- `find Adempiere -name *.sh -exec chmod +x {} \;`
- `cd Adempiere/`
- `cp AdempiereEnvTemplate.properties AdempiereEnv.properties`
- `dos2unix AdempiereEnv.properties `

#### Edit Adempiere Server Properties File
Set the values accordingly to your installation.
- `vi AdempiereEnv.properties `
- Set at least the following properties:
    - `ADEMPIERE_HOME=/home/Adempiere/Adempiere`
    - `JAVA_HOME=/usr/lib/jvm/java-1.11.0-openjdk-amd64`
    - `ADEMPIERE_APPS_PATH=/home/adempiere/apache-tomcat-9.0.71`

#### Install ADempiere on console (Background Process)
- `./RUN_silentsetup.sh`

#### Install ADempiere With GUI (Foreground Process)
- `./RUN_setup.sh`
- On the Dialog, type in the correct data
- Pay attention to the fields 
    - _Application Server Directory_: Tomcat installation path
    - _Deployment_: Tomcat configuration path inside ADempiere
    
![Selection_502a](https://user-images.githubusercontent.com/1789408/216170042-0595b5d1-5d65-4bb5-ba01-0f6301ffe994.png)

#### Import 3.9.4 Database Seed
- `cd utils`
- `./RUN_ImportAdempiere.sh`

#### Possible errors
- Error related to encoding on command _RUN_ImportAdempiere.sh_
    - Edit _utils/PostgreSQL/ImportAdempiere.sh_ 
    - Add "**-T template0**" to the _createdb_ call
     (e.g. `PGPASSWORD=$dbPwd createdb -h $dbHost -p $dbPort -E UNICODE -O $dbUser -U $dbUser -T template0 $dbName`)

### Run ADempiere on Tomcat
- `./RUN_Server2.sh`
- Open a browser.
  Adempiere should then be accessible on `http://<server-ip>:8888`
- Server logs at _/home/adempiere/Adempiere/tomcat/logs/catalina.out_

**Comment by sdgathman** [comment]
vim will convert the line endings without resorting to dos2unix.

**Comment by Winseslas** [comment]
Here's the problem I'm having




*** 2023-10-03 17:05:31.489 Adempiere Log (CLogFile) ***
17:05:31.479   CLogMgt.setLevel: CONFIG [1]
17:05:32.624 Setup.<init>: ADempiere(r) Release 3.9.4_2023-01-24 - Smart Suite ERP,CRM and SCM - (c) 1999-2023 ADempiere(r) Implementation: "ADempiere Foundation Release 3.9.4" 20230124-2031 - =ADempiere Foundation [1]
17:05:32.878 ConfigurationData.load: /opt/adempiere/AdempiereEnv.properties [1]
-----------> ConfigurationData.setEncryptionType: Invalid EncryptionType=null [1]
-----------> ConfigurationData.setAuthMechanism: Invalid AuthenticationMechanism=null [1]
-----------> ConfigurationData.setProtocol: Invalid Protocol=null [1]
17:05:35.776 Config.test: OK: JavaHome=/usr/lib/jvm/java-17-openjdk-amd64 [24]
17:05:35.779 ConfigurationData.testAdempiere: OK: AdempiereHome = /opt/adempiere [24]
17:05:35.799 KeyStoreMgt.<init>: /opt/adempiere/keystore/myKeystore [24]
17:05:35.848 KeyStoreMgt.createCertificate:  [24]
-----------> ConfigurationData.test: No Key Store [24]
===========> ConfigurationPanel$1.finished: KeyStore Password 
(No Key Store) [20]

*** 2023-10-03 17:05:42.174 Adempiere Log (CLogFile) ***

**Reply by Winseslas** [reply]
Voici le contenu de fichier AdempiereEnv.properties 



#Adempiere
#Tue Oct 03 17:05:45 WAT 2023
ADEMPIERE_HOME=/opt/adempiere
ADEMPIERE_KEYSTORECODEALIAS=adempiere
ADEMPIERE_DB_USER=adempiere
AutoCommit=0607d09432b51724
ADEMPIERE_MAIL_SERVER=0.0.0.0
ADEMPIERE_DB_PASSWORD=adempiere
ADEMPIERE_APPS_PATH=/opt/tomcat/apache-tomcat-9.0.73
ADEMPIERE_APPS_TYPE=tomcat
CacheWindow=0607d09432b51724
UILookFeel=0c55dfec0c465c5cb0d8e08f9070a667
ADEMPIERE_WEB_PORT=8888
ADEMPIERE_JAVA_TYPE=openJDK
ADEMPIERE_JAVA_OPTIONS=-Dfile.encoding\=UTF-8 -Xms128m -Xmx1024m --add-exports java.base/jdk.internal.misc\=ALL-UNNAMED --add-opens java.base/java.lang\=ALL-UNNAMED
ShowTrl=d504ada4fe01fe9b
ApplicationPassword=b6b0481f7abaeba718dc2d7909f382d0
ADEMPIERE_JNP_PORT=1099
SingleInstancePerWindow=d504ada4fe01fe9b
ADEMPIERE_VERSION=ADempiere Foundation Release 3.9.4
DataSource=25149d4b9a47f4def17237ffbe3f8d5e6dc2facaa9e0623c
ADEMPIERE_CERT_CN=0.0.0.0
ADEMPIERE_FTP_PREFIX=my
AutoLogin=d504ada4fe01fe9b
ADEMPIERE_DB_TYPE=PostgreSQL
ADEMPIERE_CERT_ORG=ADempiere Bazaar
LogMigrationScript=d504ada4fe01fe9b
ADEMPIERE_DB_SERVER=localhost
CDate=8ce64593a66480528bf7e28c9b6fbae895b241a400711293
ADEMPIERE_APPS_SERVER=0.0.0.0
Warning=Do_not_change_any_of_the_data_as_they_will_have_undocumented_side_effects.
WarningD=Einstellungen_nicht_aendern,_da_diese_undokumentierte_Nebenwirkungen_haben.
ADEMPIERE_FTP_PASSWORD=user@host.com
UITheme=0c55dfec0c465c5ceb3bd8ac7cb13235
ADEMPIERE_DB_NAME=adempiere
StorePassword=0607d09432b51724
JAVA_HOME=/usr/lib/jvm/java-17-openjdk-amd64
ADEMPIERE_CERT_ORG_UNIT=AdempiereUser
ADEMPIERE_KEYSTOREWEBALIAS=adempiere
ADEMPIERE_CERT_COUNTRY=US
ShowAcct=0607d09432b51724
ADEMPIERE_CERT_LOCATION=myTown
Language=37872e6aaf2596f0
ADEMPIERE_CERT_STATE=CA
TraceFile=d504ada4fe01fe9b
PrintPreview=d504ada4fe01fe9b
ValidateConnectionOnStartup=d504ada4fe01fe9b
ADEMPIERE_MAIL_USER=
ADEMPIERE_KEYSTORE=keystore/myKeystore
ADEMPIERE_DB_PATH=PostgreSQL
IMPLEMENTATION_VENDOR=ADempiere Foundation
ADEMPIERE_FTP_SERVER=0.0.0.0
TempDir=ebed1f15762a7912
AutoNew=d504ada4fe01fe9b
ApplicationUserID=b6b0481f7abaeba718dc2d7909f382d0
LoadTabMetaDataBackground=d504ada4fe01fe9b
Connection=fc7ff8bcdd51cbd87828c3621d37a2ca73d7185e0b1afabf9272773fc312aeb7413d64856889aaa1cc6c3cb9d0ccbf47c024dc13b147b2f898a324de1da962b218308018840797dc9152799af6d822b70e438aba1206188fcb7ad004f9ff3394a22a25d3617657233531bbd367865b8f93692cede19cac817e358f4d729cfd08fd86766403ae38c96503ce66ee2aef3af077fd72af745005f2024d0c94de07973faf4900fccf04c6f4642888be5acd35bd15f7a3f3c8894d3e7ccfaf5deaa939bf0d4bac082df65566422828346006f3
Charset=79cf4e2e0345fd0e
ADEMPIERE_ADMIN_EMAIL=admin@host.com
IMPLEMENTATION_VERSION=3.9.4
ADEMPIERE_KEYSTOREPASS=myPassword
ADEMPIERE_FTP_USER=anonymous
OpenWindowMaximized=d504ada4fe01fe9b
ADEMPIERE_DB_PORT=5432
ADEMPIERE_SSL_PORT=4444
ADEMPIERE_MAIL_PASSWORD=
ShowAdvanced=0607d09432b51724
ADEMPIERE_DB_SYSTEM=postgres
AdempiereSys=d504ada4fe01fe9b
TraceLevel=637ee83fbd8595d0

**Comment by piracio** [comment]
Hello, you download the AD from what URL? 
I will try to replicate your issue on my Infra. 
I know you are using the Ubuntu version 22.04 ( I will assume with patches ).

**Comment by Winseslas** [comment]
Voici le lien que j’ai utilisé pour le télécharger

   - wget
   https://github.com/adempiere/adempiere/releases/download/3.9.4/Adempiere_394LTS.tar.gz


Le mar. 3 oct. 2023 à 17:15, Horacio Miranda ***@***.***> a
écrit :

**Comment by Winseslas** [comment]
Here's the link I used to download it
wget
https://github.com/adempiere/adempiere/releases/download/3.9.4/Adempiere_394LTS.tar.gz

Le mar. 3 oct. 2023 à 17:15, Horacio Miranda ***@***.***> a
écrit :

**Comment by piracio** [comment]
Thanks, I will take a look in a few hours to reproduce your error and provide a work around as soon as possible.

**Comment by e-Evolution** [comment]
@Winseslas where unpack the distribution , the normal to unpack into the /opt/Adempiere?

kind regards 
Victor

**Comment by amirsaifi** [comment]
./RUN_silentsetup.sh is showing error 
Install ADempiere Server
JAVA_HOME is not set.
You may not be able to start the Setup
Set JAVA_HOME to the directory of your local JDK.
===================================
Starting Setup ...
===================================
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
java.util.logging.ErrorManager: 4: writer
java.io.FileNotFoundException: /home/adempiere/Adempiere/2024-02-02_4.log (Permission denied)
        at java.base/java.io.FileOutputStream.open0(Native Method)
        at java.base/java.io.FileOutputStream.open(FileOutputStream.java:298)
        at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:237)
        at java.base/java.io.FileWriter.<init>(FileWriter.java:113)
        at org.compiere.util.CLogFile.initialize(CLogFile.java:106)
        at org.compiere.util.CLogFile.<init>(CLogFile.java:71)
        at org.compiere.install.SilentSetup.main(SilentSetup.java:65)
*** 2024-02-02 07:36:21.205 Adempiere Log (CLogConsole) ***
07:36:21.199   CLogMgt.setLevel: CONFIG [1]
07:36:21.237 ConfigurationData.load: /home/adempiere/Adempiere/AdempiereEnv.properties [1]
07:36:21.279 Config.test: OK: JavaHome=/usr/lib/jvm/java-1.11.0-openjdk-amd64 [1]
-----------> ConfigurationData.test: Not found: AdempiereHome = /home/Adempiere/Adempiere [1]

Warning: One or more of the configuration test failed.

07:36:21.356   SecureEngine.<init>: org.compiere.util.Secure initialized - Secure[javax.crypto.Cipher@cad498c] [1]
-----------> Ini.loadProperties: /home/Adempiere/Adempiere/Adempiere.properties not found [1]
07:36:21.360   Ini.loadProperties: /home/Adempiere/Adempiere/Adempiere.properties [1]
===========> Ini.saveProperties: Cannot save Properties to /Adempiere/Adempiere.properties - java.io.FileNotFoundException: /Adempiere/Adempiere.properties (Permission denied) [1]

===========> Ini.saveProperties: Cannot save Properties to /Adempiere/Adempiere.properties - java.io.FileNotFoundException: /Adempiere/Adempiere.properties (Permission denied) [1]
07:36:21.389 Ini.loadProperties: /home/Adempiere/Adempiere/Adempiere.properties #28 [1]
===========> ConfigurationData.save: Cannot save Properties to /home/Adempiere/Adempiere/AdempiereEnv.properties - java.io.FileNotFoundException: /home/Adempiere/Adempiere/AdempiereEnv.properties (No such file or directory) [1]

===========> ConfigurationData.save: Cannot save Properties to /home/Adempiere/Adempiere/AdempiereEnv.properties - java.io.FileNotFoundException: /home/Adempiere/Adempiere/AdempiereEnv.properties (No such file or directory) [1]
Error Saving File

*** 2024-02-02 07:36:21.471 Adempiere Log (CLogConsole) ***
Install ADempiere Server
./RUN_silentsetup.sh: 8: .: Can't open ./myEnvironment.sh

**Reply by amirsaifi** [reply]
Dear Community,
Any clue. Couldn't trouble shoot the above error while deploying ADempoiere on ubuntu

**Comment by amirsaifi** [comment]
Successfully deployed ADempiere 3.9.4 and started Server with following message

ubuntu@ip-172-31-16-21:/home/adempiere/Adempiere/utils$ sudo ./RUN_Server2.s
h
Setting myEnvironment ....
CATALINA_BASE: /home/adempiere/Adempiere/tomcat
ADempiere Server tomcat starting ...
Using CATALINA_BASE:   /home/adempiere/Adempiere/tomcat
Using CATALINA_HOME:   /home/adempiere/apache-tomcat-9.0.71
Using CATALINA_TMPDIR: /home/adempiere/Adempiere/tomcat/temp
Using JRE_HOME:        /usr/lib/jvm/java-1.11.0-openjdk-amd64
Using CLASSPATH:       /home/adempiere/apache-tomcat-9.0.71/bin/bootstrap.jar:/home/adempiere/Adempiere/tomcat/bin/tomcat-juli.jar
Using CATALINA_OPTS:
Using CATALINA_PID:    /home/adempiere/Adempiere/tomcat/tomcat.pid
Existing PID file found during start.
Removing/clearing stale PID file.
Tomcat started.

but browser showing error on while accessing ADempiere. Any idea, Instance is deployed on Amazon EC2 instance.


Type Exception Report

Message Servlet.init() for servlet [Broadcast] threw exception

Description The server encountered an unexpected condition that prevented it from fulfilling the request.

Exception

javax.servlet.ServletException: Servlet.init() for servlet [Broadcast] threw exception
	org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)
	org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:687)
	org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:360)
	org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399)
	org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:891)
	org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1784)
	org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	java.base/java.lang.Thread.run(Thread.java:829)
Root Cause

java.lang.NoClassDefFoundError: org/adempiere/legacy/apache/ecs/Element
	org.compiere.cm.HttpServletCM.init(HttpServletCM.java:161)
	org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)
	org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:687)
	org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:360)
	org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399)
	org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:891)
	org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1784)
	org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	java.base/java.lang.Thread.run(Thread.java:829)
